### PR TITLE
BCDA-4301: Manage Build Tools via Go Modules

### PR DESCRIPTION
**Pull request by katyasoup**
### Fixes [BCDA-4301](https://jira.cms.gov/browse/BCDA-4301)
Currently, we are downloading our various build tools within our Dockerfiles. When we start leveraging the go-cache when we build our containers, we will no longer update these tools (cache does not see the change).

We want to leverage the strategy documented here: [https://marcofranssen.nl/manage-go-tools-via-go-modules](https://marcofranssen.nl/manage-go-tools-via-go-modules/)

### Change Details
- Go through our various Dockerfiles and add their imports to a new `tools.go` file - This will allow us to have the `go.mod` file be the source of truth, and have changes picked up by go-cache
- Update Dockerfiles to leverage this new `tools.go` file

### Security Implications
- [x] new software dependencies
_Dependencies that were previously only imported in the Dockerfiles are now brought in to `go.mod` along with their respective indirects_
- [ ] security controls or supporting software altered
- [ ] new data stored or transmitted
- [ ] security checklist is completed for this change
- [ ] requires more information or team discussion to evaluate security implications
- [x] no PHI/PII is affected by this change

### Acceptance Validation
- tests updated and passing locally

### Feedback Requested

### CODE REVIEW

File: Dockerfiles/Dockerfile.tests [hunk 1]
```diff
@@ -1,18 +1,17 @@
 FROM golang:1.15-buster
 
 RUN curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin
+ 
+WORKDIR /go/src/github.com/CMSgov/bcda-app
```
Comment (mtrang1263) [inline Dockerfiles/Dockerfile.tests:5]: Is there caching advantages from moving this block up?

Comment (katyasoup) [reply Dockerfiles/Dockerfile.tests:5]: Nope, just moving it up so I can access the tools.go file

File: db/migrations/migrations_test.go [hunk 2]
```diff
@@ -502,41 +504,28 @@ type migrator struct {
 }
 
 func (m migrator) runMigration(t *testing.T, idx string) {
```
Comment (mtrang1263) [inline db/migrations/migrations_test.go:506]: Can we change the function signature of `runMigration` to take in the idx as a `uint`?

**Review by mtrang1263 — COMMENTED** [review]
Small comment on the runMigration signature, but overall LGTM!

**Review by mtrang1263 — DISMISSED** [review]
LGTM!

**Review by mtrang1263 — APPROVED** [review]

File: db/migrations/migrations_test.go [hunk 3]
```diff
@@ -501,42 +502,26 @@ type migrator struct {
...
-	}
-
-	args = append([]string{"-database", m.dbURL, "-path",
-		m.migrationPath}, args...)
-
-	_, err := exec.Command("migrate", args...).CombinedOutput()
-	if err != nil {
-		t.Errorf("Failed to run migration %s", err.Error())
-	}
+func (m migrator) runMigration(t *testing.T, idx uint) {
```
Comment (CosmicSyntax) [inline db/migrations/migrations_test.go:505]: 👍

**Review by CosmicSyntax — APPROVED** [review]
This LGTM!

**Pull request merged**